modules/core/src/main/scala/scala/build/internals/OsLibc.scala

```diff
@@ -74,14 +74,7 @@ object OsLibc {
   }
 
   def defaultJvm(os: String): String = {
-    val hasEmptyJavaHome = Option(System.getenv("JAVA_HOME")).exists(_.trim.isEmpty)
-    val defaultJvm0      = baseDefaultJvm(os, defaultJvmVersion)
-    if (hasEmptyJavaHome)
-      // Not using the system JVM if JAVA_HOME is set to an empty string
-      // (workaround for https://github.com/coursier/coursier/issues/2292)
```

Comment:
While we're at it, I think we should also have this workaround at `JavaOptions.scala line: 153`.

Reply:
Added nonEmpty to be on the safe side, though I don't know when it would fail, so I haven't added a test.